internal/handler/msg_getmore.go

```diff
@@ -266,3 +273,68 @@ func (h *Handler) makeNextBatch(c *cursor.Cursor, batchSize int64) (*types.Array
 
 	return nextBatch, nil
 }
+
+// awaitData stops the goroutine, and waits for a new data for the cursor.
```

Comment:
To me "stops the goroutine" phrase is a bit unclear. I don't see it stopping anything, it rather retries unless `maxTimeMS` exceeds. But having in mind that this is a temporary solution, it's not that important.